src/encrypt/mod.rs

```diff
@@ -0,0 +1,74 @@
+use dusk_bls12_381::Scalar;
+
+use std::io;
+
+/// Encapsulates an encrypted data
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EncryptedData {
+    a: Scalar,
+    b: Scalar,
+}
+
+impl EncryptedData {
+    /// Perform the encryption of a given message provided a secret and a nonce
+    pub fn encrypt(message: &Scalar, secret: &Scalar, nonce: &Scalar) -> Self {
+        let a = *message;
+        let b = secret + nonce;
+
+        Self { a, b }
+    }
+
+    /// Decrypt a previously encrypted message, provided the shared secret and nonce
+    /// used for encryption. If the decryption is not successful, `None` is returned
+    pub fn decrypt(&self, secret: &Scalar, nonce: &Scalar) -> Option<Scalar> {
+        if self.b == secret + nonce {
+            Some(self.a)
+        } else {
+            None
+        }
+    }
+}
+
+impl io::Write for EncryptedData {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, io::Error> {
+        if buf.len() < 64 {
+            return Err(io::Error::from(io::ErrorKind::UnexpectedEof));
+        }
+
+        let mut a = [0x00u8; 32];
+        let mut b = [0x00u8; 32];
+
+        let mut n = a.as_mut().write(&buf[0..32])?;
+        n += b.as_mut().write(&buf[32..64])?;
```

Comment:
Nit: you don't need to specify both ranges, since you go from the beginning and to the end, so you can simplify a bit having:
```rust
    let mut n = a.as_mut().write(&buf[..32])?;
    n += b.as_mut().write(&buf[32..])?;
```